### PR TITLE
Bump dart module and improve keybindings

### PR DESCRIPTION
**Pull request by ericdallo**
After some months coding with dart I ended with these mapping which I find it's very useful.

* Improve mappings for dart
* Bump hover and lsp-dart(add test tree feature) to latest version

### CODE REVIEW

**Review by thiagokokada — APPROVED** [review]

**Comment by ericdallo** [comment]
Ping @hlissner

File: modules/lang/dart/config.el [hunk 1]
```diff
@@ -33,15 +33,33 @@
   :defer t
   :init
   (map! :map dart-mode-map
-        :localleader
-        "r" #'flutter-run-or-hot-reload))
+         "r" #'flutter-run-or-hot-reload))
```
Comment (hlissner) [inline modules/lang/dart/config.el:36]: Did you mean to remove `:localleader` here? `r` as a floating keybind doesn't seem right.

Also, please correct the indentation here (it's one space off).

File: modules/lang/dart/packages.el [hunk 2]
```diff
@@ -4,8 +4,8 @@
 (package! dart-mode :pin "04fcd649f19d49390079fbf2920a10bf37f6a634")
 
 (when (featurep! +lsp)
-  (package! lsp-dart :pin "9036e4b3a7666162fd1a6f8ec339157c28f788f0"))
+  (package! lsp-dart :pin "cdb41f73fecec367547b5a2a27a2d96b5ad55183"))
```
Comment (hlissner) [inline modules/lang/dart/packages.el:7]: Would you mind re-bumping `lsp-dart` (and updating the commit message). Seems it's moved forward a bit while I was on break.

**Review by hlissner — CHANGES_REQUESTED** [review]

**Comment by ericdallo** [comment]
Done @hlissner

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Good work. Thanks again!

**Pull request merged**